data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/KafkaSource.java

```diff
@@ -154,13 +166,60 @@ private long calculateLongestThreadWaitingTime() {
                 orElse(1L);
     }
 
-    private Properties getConsumerProperties(TopicConfig topicConfig) {
+    public String getBootStrapServersForMsk(final AwsIamAuthConfig awsIamAuthConfig, final AwsConfig awsConfig) {
+        ClientConfiguration clientConfiguration = new ClientConfiguration();
+        AWSCredentialsProvider credentialProvider;
+        if (awsIamAuthConfig == AwsIamAuthConfig.DEFAULT) {
+            credentialProvider = new DefaultAWSCredentialsProviderChain();
+        } else if (awsIamAuthConfig == AwsIamAuthConfig.ROLE) {
+            String sessionName = "data-prepper-kafka-session"+UUID.randomUUID();
+            credentialProvider = new STSAssumeRoleSessionCredentialsProvider.Builder(awsConfig.getStsRoleArn(), sessionName).build();
+        } else {
+            throw new RuntimeException("Unknown AWS IAM auth mode");
+        }
+        AWSKafka kafkaClient = AWSKafkaClientBuilder.standard()
+                .withClientConfiguration(clientConfiguration)
+                .withCredentials(credentialProvider)
+                .withRegion(awsConfig.getRegion())
+                .build();
+        GetBootstrapBrokersRequest request = new GetBootstrapBrokersRequest();
+        String clusterArn = awsConfig.getAwsMskConfig().getArn();
+        request.setClusterArn(clusterArn);
+        GetBootstrapBrokersResult result = kafkaClient.getBootstrapBrokers( request );
+        // TODO return this based on the broker_connection_mode
+        return result.getBootstrapBrokerStringSaslIam();
+    }
+
+    private Properties getConsumerProperties(final TopicConfig topicConfig) {
         Properties properties = new Properties();
+        AwsIamAuthConfig awsIamAuthConfig = null;
+        AwsConfig awsConfig = sourceConfig.getAwsConfig();
+        if (sourceConfig.getAuthConfig() != null) {
```

Comment:
Can user configure awsAuth without sourceConfig.getAuthConfig()? or with non-SASL config. it may be better to create a separate validation class to cover all combinations.

Reply:
Yes, they could configure awsConfig without authConfig but it won't be used. On the other hand, if they configured authConfig but not awsConfig, I will have to print error and shutdown the pipeline. Let me add that.

Reply:
Check for AwsConfig being null when non-null config is expected is already there.